packages/blade/src/components/Popover/Popover.web.tsx

```diff
@@ -0,0 +1,180 @@
+import { useMergeRefs } from '~utils/useMergeRefs';
+import { makeAccessible } from '~utils/makeAccessible';
+import { useId } from '~utils/useId';
+
+const Popover = ({
+  content,
+  title,
+  titleLeading,
+  footer,
+  children,
+  placement = 'top',
+  onOpenChange,
+  zIndex = popoverZIndex,
+  isOpen,
+  defaultIsOpen,
+  initialFocusRef,
+}: PopoverProps): React.ReactElement => {
+  const { theme } = useTheme();
+  const defaultInitialFocusRef = React.useRef<HTMLButtonElement>(null);
+  const arrowRef = React.useRef<SVGSVGElement>(null);
+  const titleId = useId('popover-title');
+
+  const GAP = theme.spacing[2];
+  const [side] = getPlacementParts(placement);
+  const isHorizontal = side === 'left' || side === 'right';
+  const isOppositeAxis = side === 'right' || side === 'bottom';
+
+  const [controllableIsOpen, controllableSetIsOpen] = useControllableState({
+    value: isOpen,
+    defaultValue: defaultIsOpen,
+    onChange: (isOpen) => onOpenChange?.({ isOpen }),
+  });
+
+  const { refs, floatingStyles, context, placement: computedPlacement } = useFloating({
+    open: controllableIsOpen,
+    onOpenChange: (isOpen) => controllableSetIsOpen(() => isOpen),
+    placement,
+    strategy: 'fixed',
+    middleware: [
+      shift({ crossAxis: false, padding: GAP }),
+      flip({ padding: GAP, fallbackAxisSideDirection: 'end' }),
+      offset(GAP + ARROW_HEIGHT),
+      arrow({
+        element: arrowRef,
+        padding: isHorizontal ? 0 : ARROW_WIDTH,
+      }),
+    ],
+    whileElementsMounted: autoUpdate,
+  });
+
+  const close = React.useCallback(() => {
+    controllableSetIsOpen(() => false);
+  }, [controllableSetIsOpen]);
+
+  // we need to animate from the offset of the computed placement
+  // because placement can change dynamically based on available space
+  const [computedSide] = getPlacementParts(computedPlacement);
+  const computedIsHorizontal = computedSide === 'left' || computedSide === 'right';
+  const animationOffset = isOppositeAxis ? -size[4] : size[4];
+  const { isMounted, styles } = useTransitionStyles(context, {
+    duration: theme.motion.duration.quick,
+    initial: {
+      opacity: 0,
+      transform: `translate${computedIsHorizontal ? 'X' : 'Y'}(${animationOffset}px)`,
+    },
+  });
+
+  // remove click handler if popover is controlled
+  const isControlled = isOpen !== undefined;
+  const click = useClick(context, { enabled: !isControlled });
+  const dismiss = useDismiss(context);
+  const role = useRole(context);
+
+  const { getReferenceProps, getFloatingProps } = useInteractions([click, dismiss, role]);
+  const triggerRef = React.useRef<HTMLButtonElement>(null);
+  const mergedRef = useMergeRefs(refs.setReference, triggerRef);
+
+  const contextValue = React.useMemo(() => {
+    return {
+      close,
+      defaultInitialFocusRef,
+      titleId,
+    };
+  }, [close, titleId]);
+
+  // Inject aria attributes to trigger
+  // Doing it this way instead of makeAccessible()
```

Comment:
didn't quite understand this? what is the context here?

Reply:
FloatingUI's getReferenceProps() returns a few props which needs to be passed down to the popover's trigger element. 

These are: 
- aria-describedby
- aria-lablledby
- aria-controls
- aria-haspopup
- aria-expanded
- and the normal React event props (onClick, onMouseDown etc)


Now all these props are needed to be injected into the trigger element in order to have proper behaviour / a11y. But if we pass all of them via just spreading the props that means we will need to expose these props from all of our components, thats a bit tedious so instead of spreading them I'm directly attaching them to the DOM node via `node.setAttribute` (since these aria props are just DOM props)